tools/azsdk-cli/docs/specs/0-core-telemetry.spec.md

```diff
@@ -0,0 +1,1528 @@
+# Spec: Core - Telemetry & Dashboard Requirements
+
+## Table of Contents
+
+- [Definitions](#definitions)
+- [Background / Problem Statement](#background--problem-statement)
+- [Goals and Exceptions/Limitations](#goals-and-exceptionslimitations)
+- [Design Proposal](#design-proposal)
+- [Telemetry Schema](#telemetry-schema)
+- [System Lifecycle Events](#system-lifecycle-events)
+- [Agent Prompts](#agent-prompts)
+- [CLI Commands](#cli-commands)
+- [Dashboard Requirements](#dashboard-requirements)
+- [Open Questions](#open-questions)
+- [Success Criteria](#success-criteria)
+- [Implementation Plan](#implementation-plan)
+- [Testing Strategy](#testing-strategy)
+- [Documentation Updates](#documentation-updates)
+- [Privacy and Security Considerations](#privacy-and-security-considerations)
+
+---
+
+## Definitions
+
+**Telemetry**: The automated collection, transmission, and analysis of data from the MCP server and CLI application to measure usage, performance, reliability, and user behavior patterns.
+
+**MCP Server**: The Model Context Protocol server that provides tools and capabilities to AI agents (like GitHub Copilot) for Azure SDK development workflows.
+
+**CLI Application**: The command-line interface application (`azsdk`) that provides direct command-line access to Azure SDK tools and workflows.
+
+**Tool Invocation**: The execution of a specific MCP tool or CLI command, tracked from initiation through completion or failure.
+
+**MAU (Monthly Active Users)**: Unique users who have used the MCP server or CLI at least once during a calendar month.
+
+**MEU (Monthly Engaged Users)**: Unique users who have used the MCP server or CLI on 2 or more separate days within a calendar month.
+
+**MDU (Monthly Dedicated Users)**: Unique users who have used the MCP server or CLI on 10 or more separate days within a calendar month.
+
+**System Event**: A telemetry event related to the lifecycle of the MCP server or CLI application itself (install, start, stop, etc.) rather than specific tool invocations.
+
+**User Event**: A telemetry event triggered by explicit user actions through agent prompts or CLI commands.
+
+**DevDeviceId**: A privacy-preserving hashed identifier unique to a user's development device, used for aggregating telemetry without storing personally identifiable information.
+
+**Custom SDK**: An SDK that includes hand-written code customizations beyond the auto-generated code from TypeSpec specifications.
+
+**Standard SDK**: An SDK generated directly from TypeSpec specifications without additional custom code layers.
+
+---
+
+## Background / Problem Statement
+
+### Current State
+
+The Azure SDK Tools MCP server and CLI application currently lack comprehensive telemetry capabilities. This creates several challenges:
+
+1. **No Visibility into Usage**: We don't know who is using the tools, how often, or which features are most valuable
```

Comment:
This is not correct. We do have the telemetry for each tool. What additional info is required here?